repo/blob/beforeop/beforeop.go

```diff
@@ -0,0 +1,73 @@
+// Package beforeop implements wrapper around blob.Storage that run a given callback before all operations.
+package beforeop
+
+import (
+	"context"
+
+	"github.com/kopia/kopia/repo/blob"
+)
+
+type (
+	callback          func() error
+	onGetBlobCallback func(id blob.ID) error
+	onPutBlobCallback func(id blob.ID, opts *blob.PutOptions) error // allows mutating the put-options
+)
+
+type beforeOp struct {
+	blob.Storage
+	onGetMetadata, onDeleteBlob callback
+	onGetBlob                   onGetBlobCallback
```

Comment:
only PutBlob is used in this PR, I presume the rest is for the upgrade PR?

Reply:
correct